etl-jobs/src/main/scala/org/learner/jobs/UserLookupMigration.scala

```diff
@@ -0,0 +1,128 @@
+        .config("spark.cassandra.read.timeoutMS", config.getString("spark.cassandra.read.timeoutMS"))
+        .getOrCreate()
+
+     
+    val res = time(migrateData());
+
+    Console.println("Time taken to execute script", res._1);
+    spark.stop();
+  }
+
+def migrateData()(implicit spark: SparkSession) {
+    val userschema = Encoders.product[User].schema
+    val stateUserSchema = Encoders.product[UserExternalIdentity].schema
+    val userOrgSchema = Encoders.product[UserOrg].schema
+    val userLookupschema = Encoders.product[UserLookup].schema
+
+    // Read user, user_org and usr_external identity table intot the memory 
+
+    val userdata = spark.read.format("org.apache.spark.sql.cassandra").schema(userschema).option("keyspace", "sunbird").option("table", "user").load();
+    val userOrgData = spark.read.format("org.apache.spark.sql.cassandra").schema(userOrgSchema).option("keyspace","sunbird").option("table","user_org").load();
+    val stateUserExternalIdData = spark.read.format("org.apache.spark.sql.cassandra").schema(stateUserSchema).option("keyspace","sunbird").option("table","usr_external_identity").load();
+
+    // Filter out the user records where all of email, phone, username are null.
+
+    val filteredUserData = userdata.where(col("id").isNotNull && col("username").isNotNull || col("email").isNotNull || col("phone").isNotNull).persist(StorageLevel.MEMORY_ONLY);
+    println("user data Count : " + filteredUserData.count());
+    
+    //Remove records where userid, externalid are null and is not a state users
+
+    val filteredStateUserData =stateUserExternalIdData.where(col("userid").isNotNull && col("externalid").isNotNull && col("idtype").isNotNull && col("provider").isNotNull && col("idtype") === col("provider")).persist(StorageLevel.MEMORY_ONLY);
+    println("State Users:" +filteredStateUserData.count());
+
+    val filteredUserOrg = userOrgData.where(col("userid").isNotNull && col("organisationid").isNotNull).persist(StorageLevel.MEMORY_ONLY);
+    println("User Org Mapping count:"+ filteredUserOrg.count());
+
+    import spark.implicits._
+    val userOrgMap = userOrgData.select(col("userid"),col("organisationid")).map(r=>(r.getAs[String](0),r.getAs[String](1))).collect.toMap;
+    val translationMap: Column = typedLit(userOrgMap)
+    val userLookupSchema = Encoders.product[UserLookup].schema;
+    val userLookupDF =  spark.read.format("org.apache.spark.sql.cassandra").schema(userLookupSchema).option("keyspace", "sunbird").option("table", "user_lookup").load();
+    
+
+    //Update value as email@orgId, phone@orgId, username@orgId, externalid@orgId
+
+    val stateUserLookupDF = filteredStateUserData.select(col("externalId"),col("userid"),col("provider")).withColumn("type",lit("externalid")).withColumn("value",concat(col("externalid"),lit("@"),col("provider"))).withColumn("userid",col("userid")).select(col("type"),col("value"),col("userid"));
```

Comment:
wouldn't this be 123@<provider> ? We wanted to be 123@<orgId>.

Reply:
it will be 123@orgiId